python/triqs_dft_tools/converters/wannier90.py

```diff
@@ -257,11 +262,33 @@ def convert_dft_input(self):
             # now grab the data from the H(R) file
             mpi.report(
                 "\nThe Hamiltonian in MLWF basis is extracted from %s files..." % file_seed)
-            (nr, rvec, rdeg, nw, hamr, u_mat, udis_mat,
-             band_mat, k_mesh_from_umat) = self.read_wannier90data(file_seed, kmesh_mode)
-            # number of R vectors, their indices, their degeneracy, number of
-            # WFs, H(R)
-            mpi.report("\n... done: %d R vectors, %d WFs found" % (nr, nw))
+            nr = rvec = rdeg = nw = hamr = u_mat = udis_mat = band_mat = k_mesh_from_umat = None
+            if (mpi.is_master_node()):
+                (nr, rvec, rdeg, nw, hamr, u_mat, udis_mat,
+                 band_mat, k_mesh_from_umat) = self.read_wannier90data(file_seed, kmesh_mode)
+            mpi.barrier()
+            nr = mpi.bcast(nr)
+            rvec = mpi.bcast(rvec)
+            rdeg = mpi.bcast(rdeg)
+            nw = mpi.bcast(nw)
+            hamr = mpi.bcast(hamr)
+            u_mat = mpi.bcast(u_mat)
+            udis_mat = mpi.bcast(udis_mat)
+            band_mat = mpi.bcast(band_mat)
+            k_mesh_from_umat = mpi.bcast(k_mesh_from_umat)
+            # number of R vectors, their indices, their degeneracy, number of WFs, H(R),
+            # U matrices, U(dis) matrices, band energies, k_mesh of U matrices
+            mpi.report('\n... done: {} R vectors, {} WFs found'.format(nr, nw))
+
+            if self.add_lambda:
```

Comment:
Could we make sure here, that this throws an error if the number of orbitals is not exactly 3? Would that make sense?

Reply:
Thanks. That's a good suggestion. I added a test to check for that.